tests/php/Model/SubmittedFileFieldTest.php

```diff
@@ -61,4 +64,34 @@ public function testDeletingSubmissionRemovesFile()
 
         $this->assertNull($liveVersion, 'Live file has been deleted');
     }
+
+    public function testGetFormattedValue()
+    {
+        $fileName = $this->submittedFile->getFileName();
+        $canViewType = $this->file->CanViewType;
+
+        $this->file->CanViewType = 'OnlyTheseUsers';
+        $this->file->write();
+        
+        $this->loginWithPermission('ADMIN');
+        $this->assertEquals(
+            sprintf(
+                '%s - <a href="http://localhost/assets/3c01bdbb26/test-SubmittedFileFieldTest.txt" target="_blank">Download File</a>',
+                $fileName
+            ),
+            $this->submittedFile->getFormattedValue()->value
+        );
+            
+        $this->loginWithPermission('CMS_ACCESS_CMSMain');
+        $this->assertEquals(
+            sprintf(
+                '<span class="grid-field__warning-message">%s - Insufficient rights to view the file</span>',
+                $fileName
+            ),
+            $this->submittedFile->getFormattedValue()->value
+        );
+
```

Comment:
I would ad an extra assertion here.

```suggestion
        $store = Injector::inst()->get(AssetStore::class);
        $this->assertFalse($store->canView($this->file->getFilename(), $this->file->getHash()));
```

Reply:
DONE